### PR TITLE
Move internal password validation roles into github

### PR DESCRIPTION
**Pull request by sutser**

### CODE REVIEW

**Review by janhohenheim — CHANGES_REQUESTED** [review]
The compressed files are a duplication of the dictionaries and shouldn't be in the repo.
1. Add them to the `.gitignore`, probably `*.cmp` will be fine
2. Remove the currently added files by runniing `git rm zxcvbn-test/*.cmp --cached`
3. Move the compression function into an own project configured for .NET Core (I don't think we'll have any benefit from making this executable on .NET Framework as well)
4. Pass the dictionary path and the output path via the command line (If you want to be fancy, you can use the brand new[ `System.CommandLine.Experimental` package](https://github.com/dotnet/command-line-api/wiki/Your-first-app-with-System.CommandLine)
5. Call the executable from step 4) before launching the tests

**Comment by sutser** [comment]
https://github.com/messerli-informatik-ag/zxcvbn-cs/pull/1#pullrequestreview-305713091 Addressed in: 541954d

**Comment by janhohenheim** [comment]
1. Remove the `packages` folder from the repo
2. The .cmp files are still there

File: DictionaryCompressor/Properties/launchSettings.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
```
Comment (janhohenheim) [inline DictionaryCompressor/Properties/launchSettings.json]: Remove this file from the repo and add it to the `.gitignore`

File: DictionaryCompressor/FileCompressor.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            using (FileStream originalFileStream = file.OpenRead())
+            {
+                if ((File.GetAttributes(file.FullName) & FileAttributes.Hidden)
+                    != FileAttributes.Hidden & file.Extension != ".cmp")
+                {
+                    output ??= file.DirectoryName;
+
+                    var compressedFileName = Path.GetFullPath(Path.Combine(output, file.Name + ".cmp"));
+                    using (FileStream compressedFileStream = File.Create(compressedFileName))
+                    {
```
Comment (janhohenheim) [inline DictionaryCompressor/FileCompressor.cs]: You can remove these braces

File: DictionaryCompressor/FileCompressor.cs [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+namespace DictionaryCompressor
+{
+    internal class FileCompressor : IFileCompressor
+    {
+        public void CompressFile(string filePath, string output)
+        {
+            FileInfo file = new FileInfo(filePath);
+            using (FileStream originalFileStream = file.OpenRead())
+            {
+                if ((File.GetAttributes(file.FullName) & FileAttributes.Hidden)
```
Comment (janhohenheim) [inline DictionaryCompressor/FileCompressor.cs]: Split this condition into multiple variables

File: DictionaryCompressor/FileCompressor.cs [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+using System.IO.Compression;
+using System.Linq;
+
+namespace DictionaryCompressor
+{
+    internal class FileCompressor : IFileCompressor
+    {
+        public void CompressFile(string filePath, string output)
+        {
+            FileInfo file = new FileInfo(filePath);
```
Comment (janhohenheim) [inline DictionaryCompressor/FileCompressor.cs]: use `var`

File: DictionaryCompressor/FileCompressor.cs [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+            using (FileStream originalFileStream = file.OpenRead())
+            {
+                if ((File.GetAttributes(file.FullName) & FileAttributes.Hidden)
+                    != FileAttributes.Hidden & file.Extension != ".cmp")
```
Comment (janhohenheim) [inline DictionaryCompressor/FileCompressor.cs]: Are you sure you want to use a bitwise AND here?

Comment (janhohenheim) [inline DictionaryCompressor/FileCompressor.cs]: Move `.cmp` into a constant

File: DictionaryCompressor/FileCompressor.cs [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+                {
+                    output ??= file.DirectoryName;
+
+                    var compressedFileName = Path.GetFullPath(Path.Combine(output, file.Name + ".cmp"));
```
Comment (janhohenheim) [inline DictionaryCompressor/FileCompressor.cs]: `.cmp` is duplicated

File: DictionaryCompressor/IFileCompressor.cs [hunk 7]
```diff
@@ -0,0 +1,9 @@
+﻿namespace DictionaryCompressor
+{
+    public interface IFileCompressor
```
Comment (janhohenheim) [inline DictionaryCompressor/IFileCompressor.cs]: Why do you pass an `output` string instead of returning a string?

File: DictionaryCompressor/FileCompressor.cs [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+                            originalFileStream.CopyTo(compressionStream);
+                        }
+                    }
+                }
+            }
+        }
+
+        public void CompressDirectory(string directoryPath, string output)
+        {
+            DirectoryInfo directorySelected = new DirectoryInfo(directoryPath);
```
Comment (janhohenheim) [inline DictionaryCompressor/FileCompressor.cs]: Use var

File: DictionaryCompressor/FileCompressor.cs [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+            directorySelected.GetFiles().ToList().ForEach(file => CompressFile(file.FullName, output));
```
Comment (janhohenheim) [inline DictionaryCompressor/FileCompressor.cs]: Split this on multiple lines

Comment (janhohenheim) [reply DictionaryCompressor/FileCompressor.cs]: Also, if you use `Messerli.Utility`, you can skip the call to `ToList()`.  However, I suspect you'll rewrite this anyways when you rewrite `CompressFile` to return a string.

File: DictionaryCompressor/DictionaryCompressor.csproj [hunk 10]
```diff
@@ -0,0 +1,20 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
```
Comment (janhohenheim) [inline DictionaryCompressor/DictionaryCompressor.csproj]: Use all settings from our projects, e.g. EnableNullable and TreatWarningsAsErrors

File: DictionaryCompressor/CommandLineParsing/Parser/CommandLineParser.cs [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+                new Option(
+                    "--output",
+                    "Output path")
+                {
+                    Argument = new Argument<string>(),
+                },
+            };
+            IEnumerable<FileSystemInfo> paths = new List<FileSystemInfo>();
+            _rootCommand.Handler = CommandHandler.Create<List<string>, List<string>, string>((files, directories, output) =>
+                {
```
Comment (janhohenheim) [inline DictionaryCompressor/CommandLineParsing/Parser/CommandLineParser.cs]: I think it would be user friendly if you printed a message if the user forgot to used neither `--files` nor `--directorie`. Is it possible to return the `--help` output somehow?

File: DictionaryCompressor/FileCompressor.cs [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+using System.Linq;
+
+namespace DictionaryCompressor
+{
+    internal class FileCompressor : IFileCompressor
+    {
+        public void CompressFile(string filePath, string output)
+        {
+            FileInfo file = new FileInfo(filePath);
+            using (FileStream originalFileStream = file.OpenRead())
```
Comment (janhohenheim) [inline DictionaryCompressor/FileCompressor.cs]: This is not unit testable, as you use the filesystem directly. Use [`Messerli.FileOpeningBuilder`](https://github.com/messerli-informatik-ag/file-opening-builder) by injection

File: DictionaryCompressor/FileCompressor.cs [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+            {
+                if ((File.GetAttributes(file.FullName) & FileAttributes.Hidden)
+                    != FileAttributes.Hidden & file.Extension != ".cmp")
+                {
+                    output ??= file.DirectoryName;
+
+                    var compressedFileName = Path.GetFullPath(Path.Combine(output, file.Name + ".cmp"));
+                    using (FileStream compressedFileStream = File.Create(compressedFileName))
```
Comment (janhohenheim) [inline DictionaryCompressor/FileCompressor.cs]: Use the `FileOpeningBuilder`

File: DictionaryCompressor/FileCompressor.cs (hunk 9 above)
Comment (janhohenheim) [inline DictionaryCompressor/FileCompressor.cs]: `Messerli.FileSystem is not on GitHub yet, so you can't use it here. At least move the call to `GetFiles` into an own interface in order to make this calss testable.`

File: DictionaryCompressor/FileCompressor.cs [hunk 14]
```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+
+namespace DictionaryCompressor
+{
+    internal class FileCompressor : IFileCompressor
+    {
+        public void CompressFile(string filePath, string output)
```
Comment (janhohenheim) [inline DictionaryCompressor/FileCompressor.cs]: File handling and compression are two concerns, so this class violates the Single Responsibility Principle. Split off a class that compresses a `Stream`.

File: Solution Items/.gitignore [hunk 15]
```diff
@@ -0,0 +1,2 @@
+﻿# Ignore compressed dictionaries
```
Comment (janhohenheim) [inline Solution Items/.gitignore:1]: It's in the `.gitignore`, so it's clear that this line ignores something. Simply describe the kind of file that is being ignored.

File: zxcvbn-test/Properties/AssemblyInfo.cs [hunk 16]
```diff
@@ -2,35 +2,10 @@
 using System.Runtime.CompilerServices;
```
Comment (janhohenheim) [inline zxcvbn-test/Properties/AssemblyInfo.cs:2]: I think you can safely remove this file

**Review by janhohenheim — CHANGES_REQUESTED** [review]
- You still haven't removed all `.cmp` files from the repo
- As we should only store the uncompressed lists, it's redundant to have a directory named `Compressed`
- Some files in the `DictionaryCompressor` project have a terminating newline, some do not. Could you use a `.editorconfig` at that level (not in the root directory) and apply its style rules locally?
- Add unit and integration tests for the `DictionaryCompressor`, including edge cases like malformed input.
- Revert as many refactorings from the original project as possible.
- Add a readme detailing all changes from the original library and how to use this library (e.g. running the dictionary compressor first)

File: DictionaryCompressor/Properties/launchSettings.json (hunk 1 above)
Comment (sutser) [reply DictionaryCompressor/Properties/launchSettings.json]: Addressed in: https://github.com/messerli-informatik-ag/zxcvbn-cs/commit/5080fe9b0e07602b4a70ec0412cc6151956ccc75

File: DictionaryCompressor/FileCompressor.cs (hunk 5 above)
Comment (sutser) [reply DictionaryCompressor/FileCompressor.cs]: Addressed in: https://github.com/messerli-informatik-ag/zxcvbn-cs/pull/1/commits/e05f2c0b9ac10e4c34b4fd19a3f459a7cb4f6983

Comment (sutser) [reply DictionaryCompressor/FileCompressor.cs]: Addressed in: https://github.com/messerli-informatik-ag/zxcvbn-cs/pull/1/commits/68f0c4673af030bb23dc8282ef9a66202c3a49ba

File: DictionaryCompressor/FileCompressor.cs (hunk 4 above)
Comment (sutser) [reply DictionaryCompressor/FileCompressor.cs]: Addressed in: https://github.com/messerli-informatik-ag/zxcvbn-cs/pull/1/commits/220c35912f8f0225a886c57782307515fa1e75ab

File: DictionaryCompressor/FileCompressor.cs (hunk 3 above)
Comment (sutser) [reply DictionaryCompressor/FileCompressor.cs]: Addressed in: https://github.com/messerli-informatik-ag/zxcvbn-cs/pull/1/commits/68f0c4673af030bb23dc8282ef9a66202c3a49ba

File: DictionaryCompressor/FileCompressor.cs (hunk 2 above)
Comment (sutser) [reply DictionaryCompressor/FileCompressor.cs]: contradicts rule: https://github.com/DotNetAnalyzers/IDisposableAnalyzers/blob/master/documentation/IDISP001.md

File: DictionaryCompressor/FileCompressor.cs (hunk 13 above)
Comment (sutser) [reply DictionaryCompressor/FileCompressor.cs]: Addressed in: https://github.com/messerli-informatik-ag/zxcvbn-cs/commit/fb7201530e2d9e5428840541795718ba6a52d009

File: DictionaryCompressor/FileCompressor.cs (hunk 14 above)
Comment (sutser) [reply DictionaryCompressor/FileCompressor.cs]: Addressed in: fb72015

File: DictionaryCompressor/FileCompressor.cs (hunk 9 above)
Comment (sutser) [reply DictionaryCompressor/FileCompressor.cs]: Addressed in: https://github.com/messerli-informatik-ag/zxcvbn-cs/commit/2e752e0cbd0b9e93ffe55dbb7667586a58587554

File: zxcvbn-test/Properties/AssemblyInfo.cs (hunk 16 above)
Comment (sutser) [reply zxcvbn-test/Properties/AssemblyInfo.cs:2]: Addressed in: https://github.com/messerli-informatik-ag/zxcvbn-cs/commit/f700b1cdd705b88d34aff086db29f48350c392d8

Comment (janhohenheim) [reply zxcvbn-test/Properties/AssemblyInfo.cs:2]: I'm sorry, I didn't realize that the `Properties` directory was already there in the original repo. Could you revert your commit? My bad.

File: zxcvbn-cs/DefaultMatcherFactory.cs [hunk 17]
```diff
@@ -19,18 +20,26 @@ class DefaultMatcherFactory : IMatcherFactory
     {
         List<IMatcher> matchers;
 
+        private IEnumerable<string> DictionaryResources()
```
Comment (janhohenheim) [inline zxcvbn-cs/DefaultMatcherFactory.cs]: Why are these changes in the `DefaultMatcherFactory`? The original library offers a nice `IMatcherFactory` that one can use to implement their own matcher factory. Move these changes into a new file and make sure the library uses our matcher factory instead of `DefaultMatcherFactory`.

File: zxcvbn-cs/DefaultMatcherFactory.cs [hunk 18]
```diff
@@ -19,18 +20,26 @@ class DefaultMatcherFactory : IMatcherFactory
...
+        {
+            var executingAssembly = Assembly.GetExecutingAssembly();
+            string folderName = string.Format("{0}.Resources.Folder", executingAssembly.GetName().Name);
```
Comment (janhohenheim) [inline zxcvbn-cs/DefaultMatcherFactory.cs]: Document what these changes do in the readme.

File: zxcvbn-cs/Matcher/DictionaryMatcher.cs [hunk 19]
```diff
@@ -99,7 +99,7 @@ private void CalculateEntropyForMatch(DictionaryMatch match)
         {
             // Look first to wordlists embedded in assembly (i.e. default dictionaries) otherwise treat as file path
 
-            var lines = Utility.GetEmbeddedResourceLines("Zxcvbn.Dictionaries.{0}".F(wordListFile)) ?? File.ReadAllLines(wordListFile);
+            var lines = Utility.GetEmbeddedResourceLines(wordListFile) ?? File.ReadAllLines(wordListFile);
```
Comment (janhohenheim) [inline zxcvbn-cs/Matcher/DictionaryMatcher.cs]: Document this change in the readme.

File: zxcvbn-cs/README.md [hunk 20]
```diff
@@ -87,3 +87,14 @@ http://tech.dropbox.com/?p=165
 ### Licence
 
 Since `Zxcvbn-cs` is a port of the original `Zxcvbn` the original copyright and licensing applies. Cf. the LICENSE file.
+
+
+### Changes differ from mickford/zxcvbn-cs
+
+The following changes differ from the orginal one (mickford/zxcvbn-cs - https://github.com/mickford/zxcvbn-cs)
```
Comment (janhohenheim) [inline zxcvbn-cs/README.md]:
```suggestion
The following changes differ from the [original library](https://github.com/mickford/zxcvbn-cs).:
```

File: zxcvbn-cs/README.md [hunk 21]
```diff
@@ -87,3 +87,14 @@ http://tech.dropbox.com/?p=165
...
+
+- Add keyboard layouts: 
+	Swiss German, Swiss French, Italian, German, French
+		
+- Changed date matcher logics:
```
Comment (janhohenheim) [inline zxcvbn-cs/README.md]: Your writing stile switches between imperative, present simple and past simple. Choose one.

File: zxcvbn-test/ZxcvbnTestMesserliExtended.cs [hunk 22]
```diff
@@ -0,0 +1,138 @@
...
+            Assert.IsTrue(abab.Entropy > aaaa.Entropy);
+            Assert.IsTrue(ABab.Entropy > abab.Entropy);
+        }
+
+        [TestMethod]
+        [TestCategory(CategoryUnitTest)]
+        [Priority(1)]
+        public void DictionaryMatcherTest()
+        {
+            // We test if the dictionary matcher loads the messerli dictionary
```
Comment (janhohenheim) [inline zxcvbn-test/ZxcvbnTestMesserliExtended.cs]: This should be the test name instead of a comment.

File: zxcvbn-test/ZxcvbnTestMesserliExtended.cs [hunk 23]
```diff
@@ -0,0 +1,138 @@
...
+{
+    [TestClass]
+    public class ZxcvbnTest_MesserliExtended
+    {
+        private const string CategoryUnitTest = "UnitTest";
+
+        [TestMethod]
+        [TestCategory(CategoryUnitTest)]
+        [Priority(1)]
+        public void ZxcvbnEmptyPasswordMetric()
```
Comment (janhohenheim) [inline zxcvbn-test/ZxcvbnTestMesserliExtended.cs]: Describe the intend of the code. Something like `EvaluatesEmptyPasswordWithAnEntropyOfZero`.

File: zxcvbn-test/ZxcvbnTestMesserliExtended.cs [hunk 24]
```diff
@@ -0,0 +1,138 @@
...
+
+            var result = passwordChecker.EvaluatePassword("");
+
+            Assert.AreEqual(0.0, result.Entropy);
+        }
+
+        [TestMethod]
+        [TestCategory(CategoryUnitTest)]
+        [Priority(1)]
+        public void ZxcvbnPasswordMetric()
```
Comment (janhohenheim) [inline zxcvbn-test/ZxcvbnTestMesserliExtended.cs]: Describe the intend of the code.

File: zxcvbn-test/ZxcvbnTestMesserliExtended.cs [hunk 25]
```diff
@@ -0,0 +1,138 @@
...
+
+            }
+        }
+
+        [TestMethod]
+        [TestCategory(CategoryUnitTest)]
+        [Priority(1)]
+        public void SwissSpatialMatcherTest()
+        {
+            // We test if the pattern matcher matches swiss keyboards
```
Comment (janhohenheim) [inline zxcvbn-test/ZxcvbnTestMesserliExtended.cs]: Useless comment

File: zxcvbn-test/ZxcvbnTestMesserliExtended.cs [hunk 26]
```diff
@@ -0,0 +1,138 @@
...
+                Assert.AreEqual(1, dictionaryMatch.UppercaseEntropy);
+
+
+            }
+        }
+
+        [TestMethod]
+        [TestCategory(CategoryUnitTest)]
+        [Priority(1)]
+        public void SwissSpatialMatcherTest()
```
Comment (janhohenheim) [inline zxcvbn-test/ZxcvbnTestMesserliExtended.cs]: Test name should be a description.

File: zxcvbn-test/ZxcvbnTestMesserliExtended.cs [hunk 27]
```diff
@@ -0,0 +1,138 @@
...
+                    Assert.AreEqual(pattern.Length - 1, match.j);
+                    Assert.AreEqual(pattern, match.Token);
+                }
+            }
+        }
+
+        [TestMethod]
+        [TestCategory(CategoryUnitTest)]
+        [Priority(1)]
+        public void FrenchSpatialMatcherTest()
```
Comment (janhohenheim) [inline zxcvbn-test/ZxcvbnTestMesserliExtended.cs]: Test name should be a description.

File: zxcvbn-test/ZxcvbnTestMesserliExtended.cs [hunk 28]
```diff
@@ -0,0 +1,138 @@
...
+        {
+            // We test if the pattern matcher matches swiss keyboards
+            var frenchKeyboardPatterns = new System.Collections.Generic.List<string> { "*ùmlkj", "bn,;:!" };
```
Comment (janhohenheim) [inline zxcvbn-test/ZxcvbnTestMesserliExtended.cs]: Import this type.

File: zxcvbn-cs/Matcher/DateMatcher.cs [hunk 29]
```diff
@@ -0,0 +1,312 @@
...
+{
+    /// <summary>
+    /// <para>This matcher attempts to guess dates, with and without date separators. e.g. 1197 (could be 1/1/97) through to 18/12/2015.</para>
+    /// 
+    /// <para>The format for matching dates is quite particular, and only detected years in the range 00-99 and 1900-2019 are considered by
+    /// this matcher.</para>
+    /// </summary>
+    public class DateMatcher : IMatcher
+    {
+        // TODO: This whole matcher is a rather messy but works (just), could do with a touching up. In particular it does not provide matched date details for dates without separators
```
Comment (janhohenheim) [inline zxcvbn-cs/Matcher/DateMatcher.cs:17]: Indeed, this class is complete garbage. Remove this comment and add an issue to the repo. You don't need to clean this mess, it's okay.

**Review by janhohenheim — CHANGES_REQUESTED** [review]

File: zxcvbn-test/ZxcvbnTestMesserliExtended.cs (hunk 28 above)
Comment (sutser) [reply zxcvbn-test/ZxcvbnTestMesserliExtended.cs]: Addressed in: https://github.com/messerli-informatik-ag/zxcvbn-cs/pull/1/commits/d1c2b15d6614e86b295ed96549f592061148e085

File: zxcvbn-test/ZxcvbnTestMesserliExtended.cs (hunk 25 above)
Comment (sutser) [reply zxcvbn-test/ZxcvbnTestMesserliExtended.cs]: Addressed in: https://github.com/messerli-informatik-ag/zxcvbn-cs/pull/1/commits/533e3068b1feab4aa0a4f6866519c1847a6da367

File: zxcvbn-test/ZxcvbnTestMesserliExtended.cs (hunk 28 above)
Comment (sutser) [reply zxcvbn-test/ZxcvbnTestMesserliExtended.cs]: Addressed in: https://github.com/messerli-informatik-ag/zxcvbn-cs/pull/1/commits/49b4d6adbe9b58c32c782b87f43946d10f108e6f

File: zxcvbn-test/ZxcvbnTestMesserliExtended.cs (hunk 27 above)
Comment (sutser) [reply zxcvbn-test/ZxcvbnTestMesserliExtended.cs]: Addressed in: https://github.com/messerli-informatik-ag/zxcvbn-cs/pull/1/commits/d4d09df434d83829d4248d91fbd46fdff7dbff63

File: zxcvbn-test/ZxcvbnTestMesserliExtended.cs (hunk 26 above)
Comment (sutser) [reply zxcvbn-test/ZxcvbnTestMesserliExtended.cs]: Addressed in: https://github.com/messerli-informatik-ag/zxcvbn-cs/pull/1/commits/d4d09df434d83829d4248d91fbd46fdff7dbff63

File: zxcvbn-test/ZxcvbnTestMesserliExtended.cs (hunk 22 above)
Comment (sutser) [reply zxcvbn-test/ZxcvbnTestMesserliExtended.cs]: Addressed in: https://github.com/messerli-informatik-ag/zxcvbn-cs/pull/1/commits/0e2ffddf6e997dae943ad50fe63e892d0ed5d24f

File: zxcvbn-test/ZxcvbnTestMesserliExtended.cs (hunk 23 above)
Comment (sutser) [reply zxcvbn-test/ZxcvbnTestMesserliExtended.cs]: Addressed in: https://github.com/messerli-informatik-ag/zxcvbn-cs/pull/1/commits/7e5beed7364e02b3a814352daba567ce8844b2fb

File: zxcvbn-test/ZxcvbnTestMesserliExtended.cs (hunk 24 above)
Comment (sutser) [reply zxcvbn-test/ZxcvbnTestMesserliExtended.cs]: Addressed in: https://github.com/messerli-informatik-ag/zxcvbn-cs/pull/1/commits/7e5beed7364e02b3a814352daba567ce8844b2fb

File: zxcvbn-cs/DefaultMatcherFactory.cs (hunk 18 above)
Comment (sutser) [reply zxcvbn-cs/DefaultMatcherFactory.cs]: Addressed in: https://github.com/messerli-informatik-ag/zxcvbn-cs/pull/1/commits/c37b3a0cf43fa2939e8a8402940eaf8dd927b1b2

File: zxcvbn-cs/README.md (hunk 21 above)
Comment (sutser) [reply zxcvbn-cs/README.md]: Addressed in: https://github.com/messerli-informatik-ag/zxcvbn-cs/pull/1/commits/8f3cf4fda0d758466432e3b7f93d5982338047c9

File: zxcvbn-cs/DefaultMatcherFactory.cs (hunk 17 above)
Comment (sutser) [reply zxcvbn-cs/DefaultMatcherFactory.cs]: Addressed in: https://github.com/messerli-informatik-ag/zxcvbn-cs/pull/1/commits/eee61f5ef53d64b2264891cf37de8b9eb2bf8e15

File: zxcvbn-cs/Matcher/DictionaryMatcher.cs (hunk 19 above)
Comment (sutser) [reply zxcvbn-cs/Matcher/DictionaryMatcher.cs]: Addressed in: eee61f5

File: zxcvbn-test/ZxcvbnTestMesserliExtended.cs (hunk 26 above)
Comment (dngroth) [reply zxcvbn-test/ZxcvbnTestMesserliExtended.cs]: Name it `MatchesSwissGermanKeyboardPattern`

File: zxcvbn-test/ZxcvbnTestMesserliExtended.cs (hunk 27 above)
Comment (dngroth) [reply zxcvbn-test/ZxcvbnTestMesserliExtended.cs]: Same as swiss german

File: zxcvbn-cs/DefaultMatcherFactory.cs (hunk 17 above)
Comment (sutser) [reply zxcvbn-cs/DefaultMatcherFactory.cs]: Addressed in: https://github.com/messerli-informatik-ag/zxcvbn-cs/pull/1/commits/d626983b425765af87bc9127d58cecbb850dcb93

File: zxcvbn-test/ZxcvbnTestMesserliExtended.cs (hunk 26 above)
Comment (sutser) [reply zxcvbn-test/ZxcvbnTestMesserliExtended.cs]: Addressed in: https://github.com/messerli-informatik-ag/zxcvbn-cs/pull/1/commits/89ba0ebcc87ad7e63cec9c235b089d1bc5ab3fa0

File: zxcvbn-test/ZxcvbnTestMesserliExtended.cs (hunk 27 above)
Comment (sutser) [reply zxcvbn-test/ZxcvbnTestMesserliExtended.cs]: Addressed in: https://github.com/messerli-informatik-ag/zxcvbn-cs/pull/1/commits/89ba0ebcc87ad7e63cec9c235b089d1bc5ab3fa0

File: zxcvbn-cs/Matcher/DateMatcher.cs (hunk 29 above)
Comment (sutser) [reply zxcvbn-cs/Matcher/DateMatcher.cs:17]: Addressed in: https://github.com/messerli-informatik-ag/zxcvbn-cs/pull/1/commits/feb326236c456372caeffbe65b5ac0c4bf780d96

File: zxcvbn-test/ZxcvbnTestMesserliExtended.cs (hunk 22 above)
Comment (dngroth) [reply zxcvbn-test/ZxcvbnTestMesserliExtended.cs]: name `LoadsMesserliDictionary()`

Comment (sutser) [reply zxcvbn-test/ZxcvbnTestMesserliExtended.cs]: Addressed in: https://github.com/messerli-informatik-ag/zxcvbn-cs/pull/1/commits/e6cb111a3cc6f41eb627b156c515c59ab810ee9e

**Review by janhohenheim — APPROVED** [review]
